### DISCUSSION TITLE
Spatial coherence evaluation for field emitted by cavity structure

### DISCUSSION THREAD

**Discussion by kalosu** [NanoComp/meep#2836] [Q&A]
Hello there friends from the meep community,

I am interested in evaluating the spatial coherence properties of the field generated by a cavity structure.

In essence, I am interested in knowing how coherent is the wavefront being generated by my structure. 

The people working on coherent optics, model these coherent properties of fields via the complex degree of spatial coherence. 

I wonder if this concept can be applied to the field generated by my cavity in order to evaluate how "coherent" is the total field being emitted. 

Does this make any sense to you? Is this something that could be directly applied to the field generated from a FDTD simulation?

In advance, thanks a lot for the feedback and comments.

**Comment by stevengj** [comment]
Yes, but for this to make sense you have to have some incoherence/randomness in the sources.

In particular, if you have random sources drawn from some statistical distribution, e.g. uncorrelated dipole emitters (modeling spontaneous emission) in some region of space, then you can ask what the spatial correlation function is for the resulting emitted fields (at a given frequency).   This can be done using FDTD (or any other EM solver, in principle), where with FDTD you would look at the Fourier-transformed fields.

In practice, you probably want to think carefully about how you do it.  The brute-force method would be a Monte-Carlo approach, where you do zillions of random simulations, compute the resulting fields, and then compute their correlation statistics.   However, this is quite slow, and there are generally better ways to compute correlation functions of fields using deterministic methods.  (e.g. we did a lot of this sort of thing for Casimir-force calculations, which are all about correlation functions of fields generated by random quantum sources.)

In general, the correlation function of the fields is the expectation value of a quadratic function of the fields, and we have [a recent paper](http://doi.org/10.1007/s00158-022-03389-5) on efficient methods to compute such expectation values.  However, we didn't explicitly talk about computing correlation funcitions, since in that paper we were more interested in other quadratic functions (power and energy).  But hopefully it will give you some inspiration.

**Comment by oskooi** [comment]
> I am interested in knowing how coherent is the wavefront being generated by my structure.

Is this because you want to convert the scattered/radiated fields from the Meep/full-wave simulation into a series of planewaves (or rays) for importing into a ray tracer?

**Comment by kalosu** [comment]
Hi @oskooi,

Exactly. This is precisely the reason why I want to know about this. For a fully coherent field, I should be able to use a single wavefront to represent the field, for which the ray directions are perpendicular to the wavefront surface. At the beginning I thought that the fact that I have a cavity would imply that the light scattered by such a cavity could be spatially partially coherent (due to the finite size of the cavity). However, the real source of my model is a single dipole sitting at the origin of the coordinate system. 

As far as I understand it, for this case, since the scattered field originates from the same single dipole element, there is a fixed and deterministic relationship between different parts of the field after the cavity, and as such, the field is fully coherent.

Maybe you can comment more on this?

Also, can you recommend a strategy which can be used to convert from the extracted far-field information into a set of rays which can be imported into a ray-tracing software?

**Comment by stevengj** [comment]
> since the scattered field originates from the same single dipole element, there is a fixed and deterministic relationship between different parts of the field after the cavity, and as such, the field is fully coherent.

Spatially coherent, yes.  Won't be temporally coherent if the dipole is stochastic (e.g. spontaneous emission), but that doesn't matter if you consider each frequency separately.

**Comment by stevengj** [comment]
(Even if you have an ensemble of uncorrelated dipole sources, at large distances it should effectively act like a single random point source emitter.)
